### PR TITLE
Fix configuration metadata

### PR DESCRIPTION
**Pull request by saragluna**
This PR will fix the configuration metadata for Spring Cloud Azure configuration properties.
- Remove unnecessary @NestedConfigurationProperty annotations for static inner classes.
- Add missing javadocs for fields.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java [hunk 1]
```diff
@@ -53,7 +53,6 @@
     /**
      * Default UserGroup configuration.
      */
-    @NestedConfigurationProperty
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java:56]: Why delete this?
For such kind of changes, could you please add more description in the PR description? Or add more description by `Start a review` by yourself.

image

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java:56]: for static inner classes, there's no need to add @NestedConfigurationProperty

**Review by chenrujun — APPROVED** [review]
LGTM.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**